components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -663,7 +1015,13 @@ private List<RoleV2> getScimRolesList(List<Role> roles, List<String> requiredAtt
             scimRole.setDisplayName(role.getName());
             scimRole.setId(role.getId());
             scimRole.setSchemas();
-            scimRole.setLocation(SCIMCommonUtils.getSCIMRoleV2URL(role.getId()));
+
+            if (scimUrl.equals(SCIMCommonConstants.ROLES_V2)) {
+                scimRole.setLocation(SCIMCommonUtils.getSCIMRoleV2URL(role.getId()));
+            } else if(scimUrl.equals(SCIMCommonConstants.ROLES_V3)) {
```

Comment:
This pattern `<variable>.equals(<constant>)` cause for NPE
Let's use the pattern `<constant>.equals(<variable>)`